### PR TITLE
fix(ci) use our official dockerfiles for testing

### PR DESCRIPTION
**Pull request by hutchic**
using the official dockerfiles for testing de-dupes so we're certain we're testing what we're releasing. It also makes daily docker images available for publishing

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -37,6 +37,7 @@ KONG_GMP_VERSION ?= `grep KONG_GMP_VERSION $(KONG_SOURCE_LOCATION)/.requirements
 KONG_NETTLE_VERSION ?= `grep KONG_NETTLE_VERSION $(KONG_SOURCE_LOCATION)/.requirements | awk -F"=" '{print $$2}'`
 OPENRESTY_PATCHES ?= 1
 LIBYAML_VERSION ?= 0.2.1
+DOCKER_KONG_VERSION ?= chore/remove-builder-pattern
```
Comment (hutchic) [inline Makefile]: todo: merge this and then pin this to `master` I guess

File: Jenkinsfile [hunk 2]
```diff
@@ -1,7 +1,7 @@
 pipeline {
     agent none
     environment {
-        KONG_SOURCE = "next"
+        KONG_SOURCE = "master"
```
Comment (hutchic) [inline Jenkinsfile]: pending the htest fix making it to `next`

File: Makefile [hunk 3]
```diff
@@ -310,7 +311,8 @@ ifneq ($(RESTY_IMAGE_BASE),src)
 	KONG_ADMIN_URI="http://$(TEST_HOST):$(TEST_ADMIN_PORT)" \
 	KONG_PROXY_URI="http://$(TEST_HOST):$(TEST_PROXY_PORT)" \
 	TEST_COMPOSE_PATH=$(TEST_COMPOSE_PATH) \
-	KONG_BASE_IMAGE_NAME=$(DOCKER_REPOSITORY):openresty-$(RESTY_IMAGE_BASE)-$(RESTY_IMAGE_TAG)-$(DOCKER_OPENRESTY_SUFFIX) \
+	DOCKER_GO_BUILDER=$(DOCKER_REPOSITORY):openresty-$(RESTY_IMAGE_BASE)-$(RESTY_IMAGE_TAG)-$(DOCKER_OPENRESTY_SUFFIX) \
+	KONG_BASE_IMAGE_NAME=$(KONG_TEST_IMAGE_NAME) \
```
Comment (gszr) [inline Makefile]: I believe this is no longer used.

File: test/tests/03-go-plugins/up.sh [hunk 4]
```diff
@@ -12,6 +12,7 @@ docker cp go-plugin-builder-container:/go-plugins/go-hello.so .
 docker rm -f go-plugin-builder-container
 rm -rf go-plugins
 
+echo $KONG_BASE_IMAGE_NAME
```
Comment (gszr) [inline test/tests/03-go-plugins/up.sh]: This can be nuked

File: Makefile [hunk 5]
```diff
@@ -348,8 +351,10 @@ endif
 cleanup-tests:
 ifneq ($(RESTY_IMAGE_BASE),src)
 	docker-compose -f test/kong-tests-compose.yaml down
+	docker-compose -f test/kong-tests-compose.yaml rm -f
+	docker-compose -f docker-compose.yml down
```
Comment (gszr) [inline Makefile]: Is `docker-compose.yml` used anywhere? Not sure this down and rm belong here in this target.

File: Makefile (hunk 3 above)
Comment (hutchic) [reply Makefile]: `DOCKER_GO_BUILDER` is the openresty image we build go in
`KONG_BASE_IMAGE_NAME` is the Kong docker image built from our docker-kong Dockerfile

the `03-go-plugins/up.sh` builds the go plugin using one then uses the other to cp the go plugin into and test it

File: Makefile (hunk 5 above)
Comment (hutchic) [reply Makefile]: It is used and there is a fair bit of duplication between the two docker-compose files but I'll tackle that in a different PR. I tried merging them in this one and it was very challenging

File: Makefile (hunk 3 above)
Comment (gszr) [reply Makefile]: I might be missing something, but `KONG_BASE_IMAGE_NAME` isn't actually used anywhere, is it? I can only see one reference to it, and it is an `echo` that could be removed.

Comment (hutchic) [reply Makefile]: yup I see what happened I thought `KONG_BASE_IMAGE_NAME==KONG_TEST_IMAGE_NAME`

File: test/tests/03-go-plugins/up.sh (hunk 4 above)
Comment (gszr) [reply test/tests/03-go-plugins/up.sh]: @hutchic leftover

**Pull request merged**